### ISSUE TITLE
A message from the author

### ISSUE THREAD

**Issue by matiasinsaurralde** [matiasinsaurralde/facebook-tunnel#12]
Hi everyone, this is Matias, from Asunción, Paraguay, a nice country in the heart of South America.

I’ve been following all your comments regarding this tool since the beginning, but I’ve been quite busy working on other projects (just to give you an idea, in the past two years I’ve been doing software development for a TV channel, an online music teaching startup and some short term consulting jobs, at the same time…:tired_face:).

There has been **really interesting ideas**, like implementing this tool as a pluggable transport for Tor [0], replacing base64, supporting other chat plaforms, etc. Very cool stuff!
Unofortunately the Internet.org campaign is gone from my country so I’m not able to test or debug the application anymore, so I decided to write this and ask a few things.

Basically I would like to get in touch with people from countries that are still part of Internet.org (I read that Airtel Africa joined a few weeks ago!) and see if it’s viable to continue the development (with your help), so I can spend the next months performing a complete refactor, preparing documentation and finding collaborators.
I think that what can be considered a weekend hack has the potential of becoming an important tool for people with limited Internet access, so this is my biggest motivation.

Please write your comments here. Also feel free to reach me out via e-mail: matias@insaurral.de

Thank you!

P.S. If you happen to be in Paraguay, let me know :)

[0] https://trac.torproject.org/projects/tor/wiki/doc/PluggableTransports#Brainstormingfornewtransports

**Comment by 1Conan** [comment]
Philippines is still part of the internet.org campaign and I would definitely would like to try it!
The only problem is that only https://free.facebook.com is available here. Though since it's tunneled through the chat it shouldn't be a problem. Only minor simple tweaks I guess.
I'll try to port it to the free fb site if I have time

**Comment by matiasinsaurralde** [comment]
@1Conan Hey, that's interesting, actually I've started to work on a similar project, with a modular approach, just to make it easier to integrate the software with new "transports".
I'm also planning to have better documentation, tests, etc. You may find it [here](https://github.com/matiasinsaurralde/transports).
Thanks for the reply, we should chat at some point!

**Comment by 1Conan** [comment]
Only photos and videos are blocked so yeah. You can contact me thru any of this 2 emails.
admin@1conan.com / unsecphu@gmail.com

**Comment by DarkCyber5** [comment]
you have any tutorial with facebook-tunnel ? 
i need help to run facebook-tunnel on my linux so please help me with a tutorial or commands to run this.

**Comment by arthurbarros** [comment]
@matiasinsaurralde great to hear from you, I really thought this project died long back ago, but as @DarkCyber5 mentioned I think a good way to gether more colaborator would be to easy the join of not hardcore c++ developer to give a try on the project. would be possible to write up a tutorial or detail more the README on how to run it on linux based plataforms or on Mac OS X ?

Thanks a lot for this amazing project.

**Comment by psqli** [comment]
Hello, I saw this project in a post some time ago. I got very impressed and excited with the idea.
This week I've looked at the code and I thought about how could it be implemented as best as possible.
Some questions I've made to myself:
Does CAPTCHA becomes a problem if we send tons of messages?
Would be possible to send data over a file? I found wouldn't.
Is Google's Gumbo parser an efficient way to get our messages?
And to send messages ... Is there something dynamically generated by FB that can confuse us?
Unfortunately there are more things that left me hopeless by now :-/
I'm also busy with other projects however anything I can help I'll help. :-)
Cheers!

**Comment by ghost** [comment]
@1Conan have you made any software for that ? in my country also we have free basics with many free sites and facebook supports sending images too

**Comment by 1Conan** [comment]
@FaArIsH
I haven't really gotten around it since i lost interest as I would have no benefit anymore from it. 

If I get interested again, I might implement stuff in NodeJS though. i'm no good in C++ or Go
